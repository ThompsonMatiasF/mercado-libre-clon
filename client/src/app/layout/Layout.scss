.display-range {
  min-width: 768px;
  max-width: 1200px;
}
