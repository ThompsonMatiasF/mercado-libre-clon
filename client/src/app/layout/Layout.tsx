import React, {FC, ReactElement} from 'react';
import { Container, Row, Col } from 'react-bootstrap';
import './Layout.scss';

interface LayoutProps {
  header: ReactElement
}

export const Layout: FC<LayoutProps> = ({ header, children }) => (
  <Container fluid>
    <Row className="bg-warning">
      <Container className="display-range">
        <Row className="bg-yellow">
          <Col xs={{ offset: 1, span: 10 }}>{ header }</Col>
        </Row>
      </Container>
    </Row>
    <Row>
      <Container className="display-range">
        <Row>
          <Col xs={{ offset: 1, span: 10 }}>{ children }</Col>
        </Row>
      </Container>
    </Row>
  </Container>
);

export default Layout;
