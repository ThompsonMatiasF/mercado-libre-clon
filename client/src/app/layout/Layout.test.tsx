import { render } from '@testing-library/react'
import Layout from './Layout'

describe('Layout.tsx', () => {
  it('should match snapshot', () => {
    const { container } = render(<Layout header={<p>header</p>}><p>children</p></Layout>)

    expect(container).toMatchSnapshot()
  })
})
