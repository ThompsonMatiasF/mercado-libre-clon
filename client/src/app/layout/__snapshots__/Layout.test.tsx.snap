// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout.tsx should match snapshot 1`] = `
<div>
  <div
    class="container-fluid"
  >
    <div
      class="bg-warning row"
    >
      <div
        class="display-range container"
      >
        <div
          class="bg-yellow row"
        >
          <div
            class="col-10 offset-1"
          >
            <p>
              header
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="display-range container"
      >
        <div
          class="row"
        >
          <div
            class="col-10 offset-1"
          >
            <p>
              children
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
